samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/FlightBookingRecognizer.java

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.corebot.app.insights;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.StringUtils;
+import com.microsoft.bot.ai.luis.LuisApplication;
+import com.microsoft.bot.ai.luis.LuisRecognizer;
+import com.microsoft.bot.ai.luis.LuisRecognizerOptionsV3;
+import com.microsoft.bot.builder.BotTelemetryClient;
+import com.microsoft.bot.builder.Recognizer;
+import com.microsoft.bot.builder.RecognizerResult;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.integration.Configuration;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The class in charge of recognizing the booking information.
+ */
+public class FlightBookingRecognizer implements Recognizer {
+    private LuisRecognizer recognizer = null;
+
+    /**
+     * The constructor of the FlightBookingRecognizer class.
+     *
+     * @param configuration The Configuration object to use.
+     * @param telemetryClient
+     */
+    public FlightBookingRecognizer(Configuration configuration, BotTelemetryClient telemetryClient) {
+        Boolean luisIsConfigured = StringUtils.isNotBlank(configuration.getProperty("LuisAppId"))
+            && StringUtils.isNotBlank(configuration.getProperty("LuisAPIKey"))
+            && StringUtils.isNotBlank(configuration.getProperty("LuisAPIHostName"));
+
+        if (luisIsConfigured) {
+            LuisApplication luisApplication = new LuisApplication(
+                configuration.getProperty("LuisAppId"),
+                configuration.getProperty("LuisAPIKey"),
+                String.format("https://%s", configuration.getProperty("LuisAPIHostName"))
```

Comment:
```suggestion
                "https://".concat(configuration.getProperty("LuisAPIHostName"))
```

This is just an improvement, it's format to use `format` in these scenarios

Reply:
ok, we should maybe change this in sample 13 later